flang/runtime/edit-input.cpp

```diff
@@ -222,7 +222,15 @@ bool EditIntegerInput(
     value = -value;
   }
   if (any || !io.GetConnectionState().IsAtEOF()) {
-    std::memcpy(n, &value, kind); // a blank field means zero
+    // For integer kind <= 4, the value is stored in the lower order bits on
```

Comment:
It might be less confusing to instead change the definition of Fortran::common::Int128 for big-endian hosts so that its `high_` data member comes first.

Reply:
@klausler I am not sure perhaps I miss something. In here, it wants to memcpy any size less than that of `uint64_t`, the memory layout is different within the components of `Int128`, i.e. the value is in the lower order bits in BE. Swapping the order of `high` and `low` will not make things different about the layout within the 8 byte storage. The existing code assumes that the value is always in the high order bit in `low`.

Reply:
@klausler I did quickly swap the `high` and `low` in `Int128`. It causes more regressions but I haven't looked into the details.

Reply:
The case I had in mind was for 128-bit input, for which I think your code will reverse its 64-bit halves.

Reply:
If the 128-bit input, the full 128-bit will be copied. I don't think reverse is needed. It is also the case for 64-bit input. Since the value is stored in `low_`, the `memcpy` just copies from `value` for the entire `low_`. However, for 32-bit input, in big endian, the left-half of `low_` is all zero and the right-half of `low_` has the 32-bit value. The `memcpy`  will always gets zero.

Reply:
> If the 128-bit input, the full 128-bit will be copied.

Yes, that's my concern.  If the source of the copy is an instantiation of Fortran::common::Int128, the low half always comes first, so it can't just be copied into a Fortran `INTEGER(16)`.  Any built-in C++ 128-bit integer type is fine, but not Int128.  So maybe you want to modify Int128 to reflect host endianness so that it will work if it's being used in the runtime.

Reply:
I see. Yes, in fact `integer(16)` (without swapping high and low) does not work on AIX. We will update the PR with your suggestion.